egs/wsj/s5/local/nnet3/run_discriminative.sh

```diff
@@ -0,0 +1,224 @@
+# the features and the iVectors are, but of course it's a little inefficient.
+# The time taken is dominated by the lattice generation anyway, so this isn't
+# a huge deal.
+
+. cmd.sh
+
+
+stage=0
+train_stage=-10
+get_egs_stage=-10
+use_gpu=true
+srcdir=exp/nnet3/nnet_ms_a
+criterion=smbr
+drop_frames=false  # only matters for MMI.
+frames_per_eg=150
+frames_overlap_per_eg=30
+effective_learning_rate=0.0000125
+num_jobs_nnet=4
+train_stage=-10 # can be used to start training in the middle.
+decode_start_epoch=1 # can be used to avoid decoding all epochs, e.g. if we decided to run more.
+num_epochs=4
+degs_dir=
+cleanup=false  # run with --cleanup true --stage 6 to clean up (remove large things like denlats,
+               # alignments and degs).
+lats_dir=
+train_data_dir=data/train_si284_hires
+online_ivector_dir=exp/nnet3/ivectors_train_si284
+one_silence_class=true
+truncate_deriv_weights=10
+minibatch_size=64
+
+adjust_priors=true
+
+determinize=true
+minimize=true
+remove_output_symbols=true
+remove_epsilons=true
+collapse_transition_ids=true
+
+modify_learning_rates=true
+last_layer_factor=1.0
+
+set -e
+. cmd.sh
+. ./path.sh
+. ./utils/parse_options.sh
+
+
+if $use_gpu; then
+  if ! cuda-compiled; then
+    cat <<EOF && exit 1 
+This script is intended to be used with GPUs but you have not compiled Kaldi with CUDA 
+If you want to use GPUs (and have them), go to src/, and configure and make on a machine
+where "nvcc" is installed.  Otherwise, call this script with --use-gpu false
+EOF
+  fi
+  num_threads=1
+else
+  # Use 4 nnet jobs just like run_4d_gpu.sh so the results should be
+  # almost the same, but this may be a little bit slow.
+  num_threads=16
+fi
+
+if [ ! -f ${srcdir}/final.mdl ]; then
+  echo "$0: expected ${srcdir}/final.mdl to exist; first run run_tdnn.sh or run_lstm.sh"
+  exit 1;
+fi
+
+
+if [ -z "$lats_dir" ]; then
+  lats_dir=${srcdir}_denlats
+  if [ $stage -le 1 ]; then
+    nj=50  # this doesn't really affect anything strongly, except the num-jobs for one of
+    # the phases of get_egs_discriminative2.sh below.
+    num_threads_denlats=6
+    subsplit=40 # number of jobs that run per job (but 2 run at a time, so total jobs is 80, giving
+    # total slots = 80 * 6 = 480.
+    steps/nnet3/make_denlats.sh --cmd "$decode_cmd --mem 1G --num-threads $num_threads_denlats" \
+      --online-ivector-dir $online_ivector_dir \
+      --nj $nj --sub-split $subsplit --num-threads "$num_threads_denlats" --config conf/decode_dnn.config \
+      $train_data_dir data/lang $srcdir $lats_dir || exit 1;
+  fi
+fi
+
+if [ $stage -le 2 ]; then
+  # hardcode no-GPU for alignment, although you could use GPU [you wouldn't
+  # get excellent GPU utilization though.]
+  nj=100 # have a high number of jobs because this could take a while, and we might
+         # have some stragglers.
+  use_gpu=no
+  gpu_opts=
+
+  steps/nnet3/align.sh  --cmd "$decode_cmd $gpu_opts" --use-gpu "$use_gpu" \
```

Comment:
please change steps/nnet3/align.sh so that if use_gpu is yes, it automatically adds --gpu 1 to $cmd, and remove the gpu_opts variable here.